src/components/cl/hier/allgatherv/allgatherv.c

```diff
@@ -69,53 +69,48 @@ static inline ucc_status_t find_leader_rank(ucc_base_team_t *team,
    dst buffer is contiguous */
 static inline ucc_status_t is_block_ordered(ucc_cl_hier_team_t *cl_team, int *ordered)
 {
-    ucc_topo_t *topo = cl_team->super.super.params.team->topo;
-    ucc_sbgp_t *sbgp;
-    int         is_block_ordered;
+    ucc_topo_t  *topo             = cl_team->super.super.params.team->topo;
+    ucc_sbgp_t  *all_nodes        = NULL;
+    int          is_block_ordered = 1;
+    int          n_nodes;
+    ucc_status_t status;
+    ucc_rank_t   prev_rank, curr_rank;
+    ucc_rank_t   i, j;
 
     if (cl_team->is_block_ordered != -1) {
         is_block_ordered = cl_team->is_block_ordered;
     } else {
-        sbgp = ucc_topo_get_sbgp(topo, UCC_SBGP_FULL_HOST_ORDERED);
-        is_block_ordered = ucc_ep_map_is_identity(&sbgp->map) ? 1 : 0;
-        cl_team->is_block_ordered = is_block_ordered;
-    }
-
-    *ordered = is_block_ordered;
-
-    return UCC_OK;
-}
-
-/* Node leader subgroup is always ordered by ascending host_id. If the team's ranks are
-   not in the same order, then node leader subgroup allgatherv won't be enough, we'll
-   have to use a staging buffer and unpack to reorder each leader's contribution.
-   So, this func will check that the team ranks in the ldr sbgp are ascending */
-static inline ucc_status_t is_host_ordered(ucc_cl_hier_team_t *cl_team, int *ordered)
-{
-    int         is_host_ordered;
-    ucc_rank_t  max_rank, i, team_rank;
+        status = ucc_topo_get_all_nodes(topo, &all_nodes, &n_nodes);
+        if (status != UCC_OK) {
+            return status;
+        }
 
-    if (cl_team->is_host_ordered != -1) {
-        is_host_ordered = cl_team->is_host_ordered;
-    } else {
-        if (SBGP_EXISTS(cl_team, NODE_LEADERS)) {
-            is_host_ordered = 1;
-            max_rank = ucc_ep_map_eval(SBGP_MAP(cl_team, NODE_LEADERS), 0);
-            for (i = 1; i < SBGP_SIZE(cl_team, NODE_LEADERS); i++) {
-                team_rank = ucc_ep_map_eval(SBGP_MAP(cl_team, NODE_LEADERS), i);
-                if (team_rank < max_rank) {
-                    is_host_ordered = 0;
-                    break;
+        // For each node, check if its ranks are consecutive in team order
```

Comment:
Here I revert the implementation of is_block_ordered back to the original way I did it--for each node subgroup, check that the team's ranks are ascending and consecutive. The reason the old implementation doesn't work is that the full ordered subgroup also orders by socket and numa, and for the purpose of allgatherv the socket/numa placement doesn't matter.